atc/db/resource.go

```diff
@@ -156,35 +154,35 @@ func (resources Resources) Configs() atc.ResourceConfigs {
 			Source:       r.Source(),
 			CheckEvery:   r.CheckEvery(),
 			Tags:         r.Tags(),
-			Version:      r.ConfigPinnedVersion(),
+			Version:      r.PinnedVersion(),
```

Comment:
This should probably only return the `PinnedVersion()` if the version is pinned through the config. 🤔 This method is used by `pipeline.Config()` https://github.com/concourse/concourse/blob/6b3f6c8a556c2e30a9f3a1c7aed41e375fb85a52/atc/db/pipeline.go#L258
in order to fetch the config for a pipeline and we probably don't want to return a version pinned through the API as part as the pipeline configuration.